commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Messages/MessagesIntegrationTests.cs

```diff
@@ -0,0 +1,145 @@
+                Assert.Single(returnedSet.Results);
+                var customerCreatedMessage = returnedSet.Results[0];
+                Assert.NotNull(customerCreatedMessage);
+                Assert.NotNull(customerCreatedMessage.Customer);
+                Assert.Equal(customer.Id, customerCreatedMessage.Customer.Id);
+            });
+        }
+
+        [Fact]
+        public async Task QueryMultipleMessagesTypesOfSpecificResourceType()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, DefaultCustomerDraftWithAddress, async customer =>
+            {
+                Assert.Single(customer.Addresses);
+                var oldAddress = customer.Addresses[0];
+                //then change his address
+                var newAddress = TestingUtility.GetRandomAddress();
+
+                var updateActions = new List<UpdateAction<Customer>>();
+                var action = new ChangeAddressUpdateAction() {Address = newAddress, AddressId = oldAddress.Id};
+                updateActions.Add(action);
+
+                var updatedCustomer = await client
+                    .ExecuteAsync(new UpdateByIdCommand<Customer>(customer, updateActions));
+
+
+                var queryCommand = new QueryCommand<Message<Customer>>();
+                queryCommand.Where(message =>
+                    message.Type == "CustomerCreated" || message.Type == "CustomerAddressChanged");
+                queryCommand.SetLimit(2);
+                queryCommand.SetSort(new List<Sort<Message<Customer>>>()
+                {
+                    new Sort<Message<Customer>>(message => message.CreatedAt, SortDirection.Descending)
+                });
+                var returnedSet = await client.ExecuteAsync(queryCommand);
+                Assert.True(returnedSet.Results.Count > 0);
+                var customerCreatedMessage = returnedSet.Results[1] as CustomerCreatedMessage;
+                var customerAddressChangedMessage =
+                    returnedSet.Results[0] as CustomerAddressChangedMessage; // this is latest message
+
+                Assert.NotNull(customerCreatedMessage);
+                Assert.NotNull(customerCreatedMessage.Customer);
+                Assert.Equal(customer.Id, customerCreatedMessage.Customer.Id);
+                Assert.NotNull(customerAddressChangedMessage);
+                Assert.NotNull(customerAddressChangedMessage.Address);
+                Assert.Equal(updatedCustomer.Addresses[0].Id, customerAddressChangedMessage.Address.Id);
+            });
+        }
+
+        [Fact]
+        public async Task QueryDifferentMessageTypesOfDifferentResourceTypes()
+        {
+            //lets create customer first
+            await WithCustomer(client, async customer =>
+            {
+                await WithReview(client, async review =>
+                {
+                    var queryCommand = new QueryCommand<Message>();
```

Comment:
Same here with assertions and query predicate

Reply:
I did the same here